slither/tools/read_storage/__main__.py

```diff
@@ -126,22 +126,22 @@ def main() -> None:
     else:
         contracts = slither.contracts
 
-    srs = SlitherReadStorage(contracts, args.max_depth)
-
-    try:
-        srs.block = int(args.block)
-    except ValueError:
-        srs.block = str(args.block or "latest")
-
+    rpc_info = None
     if args.rpc_url:
-        # Remove target prefix e.g. rinkeby:0x0 -> 0x0.
-        address = target[target.find(":") + 1 :]
-        # Default to implementation address unless a storage address is given.
-        if not args.storage_address:
-            args.storage_address = address
-        srs.storage_address = args.storage_address
-
-        srs.rpc = args.rpc_url
+        try:
+            block = int(args.block)
+        except ValueError:
+            valid = ["latest", "earliest", "pending", "safe", "finalized"]
+            block = next((v for v in valid if v == args.block), "latest")
```

Comment:
```suggestion
            block = args.block if args.block in valid else "latest"
```

Reply:
We already default to latest for `args.block` so maybe this should throw an error as the user passed in an invalid parameter explicitly

Reply:
> We already default to latest for `args.block` so maybe this should throw an error as the user passed in an invalid parameter explicitly

Something like this?
```python
    if args.rpc_url:
        try:
            valid = ["latest", "earliest", "pending", "safe", "finalized"]
            block = args.block if args.block in valid else int(args.block)
        except ValueError as err:
            raise SlitherError(str(err)) from err
        rpc_info = RpcInfo(args.rpc_url, block)
```

Reply:
I would leave off ```
except ValueError as err:
            raise SlitherError(str(err)) from err
```

Reply:
Ok, then I'll drop the try/except and let it fail with the ValueError